File: SCT2/trunk/plugins/org.yakindu.sct.model.sgraph/src/org/yakindu/sct/model/sgraph/Choice.java
/**
 * Copyright (c) 2011 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * Contributors:
 * committers of YAKINDU - initial API and implementation
 * 
 */
package org.yakindu.sct.model.sgraph;


/**
 * <!-- begin-user-doc -->
 * A representation of the model object '<em><b>Choice</b></em>'.
 * <!-- end-user-doc -->
 *
 * <p>
 * The following features are supported:
 * <ul>
 *   <li>{@link org.yakindu.sct.model.sgraph.Choice#getKind <em>Kind</em>}</li>
 * </ul>
 * </p>
 *
 * @see org.yakindu.sct.model.sgraph.SGraphPackage#getChoice()
 * @model
 * @generated
 */
public interface Choice extends Pseudostate {
	/**
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @generated
	 */
	String copyright = "Copyright (c) 2011 committers of YAKINDU and others.\r\nAll rights reserved. This program and the accompanying materials\r\nare made available under the terms of the Eclipse Public License v1.0\r\nwhich accompanies this distribution, and is available at\r\nhttp://www.eclipse.org/legal/epl-v10.html\r\nContributors:\r\ncommitters of YAKINDU - initial API and implementation\r\n";

	/**
	 * Returns the value of the '<em><b>Kind</b></em>' attribute.
	 * The literals are from the enumeration {@link org.yakindu.sct.model.sgraph.ChoiceKind}.
	 * <!-- begin-user-doc -->
	 * <p>
	 * If the meaning of the '<em>Kind</em>' attribute isn't clear,
	 * there really should be more of a description here...
	 * </p>
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Kind</em>' attribute.
	 * @see org.yakindu.sct.model.sgraph.ChoiceKind
	 * @see #setKind(ChoiceKind)
	 * @see org.yakindu.sct.model.sgraph.SGraphPackage#getChoice_Kind()
	 * @model
	 * @generated
	 */
	ChoiceKind getKind();

	/**
	 * Sets the value of the '{@link org.yakindu.sct.model.sgraph.Choice#getKind <em>Kind</em>}' attribute.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @param value the new value of the '<em>Kind</em>' attribute.
	 * @see org.yakindu.sct.model.sgraph.ChoiceKind
	 * @see #getKind()
	 * @generated
	 */
	void setKind(ChoiceKind value);

} // Choice

File: SCT2/trunk/test-plugins/org.yakindu.sct.generator.c.test/gtests/Choice/Choice.cc
/**
* Copyright (c) 2012 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "Choice.h"

TEST(StatemachineTest, choiceTest) {
	Choice handle;
	choice_init(&handle);
	choice_enter(&handle);
	EXPECT_TRUE(choice_isActive(&handle, Choice_main_region_A));
	EXPECT_TRUE(choiceIface_get_value(&handle) == 4);
	choiceIface_raise_pressKey(&handle);
	choice_runCycle(&handle);
	EXPECT_TRUE(choice_isActive(&handle, Choice_main_region_B));
	EXPECT_TRUE(choiceIface_get_value(&handle) == 3);
	choiceIface_raise_pressKey(&handle);
	choice_runCycle(&handle);
	EXPECT_TRUE(choice_isActive(&handle, Choice_main_region_A));
	choiceIface_raise_pressKey(&handle);
	choice_runCycle(&handle);
	EXPECT_TRUE(choice_isActive(&handle, Choice_main_region_C));
	EXPECT_TRUE(choiceIface_get_value(&handle) == 2);
	choiceIface_raise_pressKey(&handle);
	choice_runCycle(&handle);
	EXPECT_TRUE(choice_isActive(&handle, Choice_main_region_B));
	EXPECT_TRUE(choiceIface_get_value(&handle) == 1);
	choiceIface_raise_pressKey(&handle);
	choice_runCycle(&handle);
	EXPECT_TRUE(choice_isActive(&handle, Choice_main_region_A));
	EXPECT_TRUE(choiceIface_get_value(&handle) == 1);
	choiceIface_raise_pressKey(&handle);
	choice_runCycle(&handle);
	EXPECT_TRUE(choice_isActive(&handle, Choice_main_region_C));
	EXPECT_TRUE(choiceIface_get_value(&handle) == 0);
	choiceIface_raise_pressKey(&handle);
	choice_runCycle(&handle);
	EXPECT_TRUE(choice_isActive(&handle, Choice_main_region_A));
}
